File: 0041-first-missing-positive/0041-first-missing-positive.py
class Solution:
    def firstMissingPositive(self, nums: List[int]) -> int:
        s = set()
        for x in nums:
            if x>0:
                s.add(x)
        
        print(s)
            
        for i in range(1, len(s) + 2):
            # print(i)
            if i not in s:
                # print('Here')
                return i 

        
        

File: 0041-first-missing-positive/0041-first-missing-positive.cpp
class Solution {
public:
    int firstMissingPositive(vector<int>& nums) {
        if(nums.size() == 1){
            if(nums[0] == 1){
                return 2;
            }
            else{
                return 1;
            }
        }
        set<int> s;
        for(auto x:nums){
            if(x>0){
                s.insert(x);
            }
        }
        auto it = s.begin();
        int i = 1;
        for(i; i<nums.size() + 1; i++){
            cout<<(*it)<<' '<<i<<endl;
            if(i != *it){
                return i;
            }
            it++;
        }
        return (i);
    }
};